modules/formulize/class/forms.php

```diff
@@ -1402,4 +1402,22 @@ public function setPermissionsForClonedForm($fid, $newfid)
 			$res = $this->db->queryF($sql);
 }
 	}
+        function hasForeignKeyLinkElement($fid){
+            global $xoopsDB;
+            $sql = "SELECT ele_foreign_key_element_link FROM " . $xoopsDB->prefix("formulize") . " WHERE id_form=" . $fid . ";";

```

Comment:
The SQL can be made a bit better by adding ele_foreign_key_element_link = 1 to the WHERE clause, then just checking if the $xoopsDB->getRowsNum($res) > 0 after running the query.